Unable to build CLR: error: assigning to 'var_types' from incompatible type 'unsigned char'

On recent commits I am unable to build the runtime repo. I get the error:

```
morph.cpp:67:35: error: assigning to 'var_types' from incompatible type 'unsigned char'
     call->gtReturnType          = static_cast<unsigned char>(tree->TypeGet());
                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This is starting to appear in CI, as well: https://dev.azure.com/dnceng/public/_build/results?buildId=1412092&view=logs&jobId=eb4db7b6-55d4-5d5f-e2fd-7706dfacdc48&j=eb4db7b6-55d4-5d5f-e2fd-7706dfacdc48&t=d73aac41-571c-5282-cbaa-a7a431519328

Build environment:

Ubuntu 20.04.2 LTS
commit 93817770ab31e50d0f9b79ffacc02041047327bf
Build steps: `./build.sh -rc release -s clr -ninja`

--- a/src/coreclr/jit/morph.cpp
+++ b/src/coreclr/jit/morph.cpp
@@ -64,7 +64,7 @@ GenTree* Compiler::fgMorphIntoHelperCall(GenTree* tree, int helper, GenTreeCall:
 
     GenTreeCall* call           = tree->AsCall();
     call->gtCallType            = CT_HELPER;
-    call->gtReturnType          = static_cast<unsigned char>(tree->TypeGet());
+    call->gtReturnType          = tree->TypeGet();
     call->gtCallMethHnd         = eeFindHelper(helper);
     call->gtCallThisArg         = nullptr;
     call->gtCallArgs            = args;

